core/src/main/java/io/questdb/cairo/GenericRecordMetadata.java

```diff
@@ -29,6 +29,20 @@
 
 public class GenericRecordMetadata extends AbstractRecordMetadata {
 
+    // Does not copy deleted columns.
+    public static GenericRecordMetadata cleanCopyOf(RecordMetadata that) {
+        if (that != null) {
+            if (that instanceof GenericRecordMetadata) {
+                return (GenericRecordMetadata) that;
+            }
+            GenericRecordMetadata metadata = new GenericRecordMetadata();
+            copyNonDeletedColumns(that, metadata);
+            metadata.setTimestampIndex(that.getTimestampIndex());
```

Comment:
Shouldn't timestamp index be shifted if there is a deleted column before it?

Reply:
All indexes are preserved as is, so no shift is necessary. `TextLoader` uses them to insert rows directly on the table writer, so the indexes must match.

Reply:
it should use`getWriterIndex` to write.

Reply:
We probably need a test that imports CSV in a table where a column before the designated timestamp is dropped.

Reply:
> it should use`getWriterIndex` to write.

This doesn't sound like a non-risky bugfix to me as it would touch quite a few classes, so I'd better do this refactoring in a separate PR.

Reply:
I changed to `copyDense` in `TextLoaderCompletedState` and tests still pass. `copyDense` shifts the timestamp. As far as I can see, this `metadata` copy is only used to print the output table in the upload response and is not involved in writing to the table.

Reply:
Moved to dense metadata in 8d9648b31595614da8344512d70fdbe984ec78f3